### ISSUE TITLE
Fail background skipped tests

### ISSUE THREAD

**Issue by rogertorres** [cucumber-rs/cucumber#198]
Currently, running a test with `fail_on_skipped()` does not fail `Background` skipped tests. The reason is because the `Background` steps are not being mapped as the steps directly under the `Scenario` are:
https://github.com/cucumber-rs/cucumber/blob/935602b2cb36be7466f96af14315fe7db03e0f84/src/writer/fail_on_skipped.rs#L92-L95
#
If that was the only thing that needed to be changed, it would be easy (_spoiler: it is not_); all it takes is to copy code above and replace `Scenario::Step` with `Scenario::Background`.
```rust
Cucumber::Feature(
    f,
    Feature::Scenario(sc, Scenario::Background(st, Step::Skipped)),
) => map_failed(f, None, sc, st),
```

It does work, but it also leads to another issue. Once we force skipped background steps to fail, we should also allow the usage of `@allow.skipped` on these steps, which is not possible because, [as you clearly stated](https://cucumber-rs.github.io/cucumber/current/writing/tags.html#failing-on-skipped-steps), we cannot use tags above `Background`.

I know you guys went deep redesigning this, so I want to understand your reasoning before proposing anything or making changes locally. My questions are:
- To fail skipped `Background` tests is not available _by design_?
- What is the reason for not allowing tags above `Background`? 

Thank you!

**Comment by ilslv** [comment]
@rogertorres thanks for detailed report and diving into code!

> To fail skipped Background tests is not available by design?

No, this definitely was overlooked. 

> It does work, but it also leads to another issue. Once we force skipped background steps to fail, we should also allow the usage of @allow.skipped on these steps, which is not possible because, as you clearly stated, we cannot use tags above Background.

Actually I don't think this is an issue, because you can still use `@allow.skipped` on all `Scenario`s that are using skipped `Background`s in case you don't want them to fail. So solution you have provided should be enough. 

I will happily review your PR in case it's possible or fix this myself.

**Comment by rogertorres** [comment]
@ilslv This makes total sense!

I submitted a minimal PR, which I tested locally. I was wondering if it would not be a good idea to add some tests and/or a tip on the book. This is something I could do, but I would then need more time/help to really understand how you organized things there.

**Comment by tyranron** [comment]
@rogertorres fix has been released in 0.11.2 version.

**Comment by rogertorres** [comment]
@tyranron Thank you!
